pkgs/development/python-modules/manticore/default.nix

```
@@ -0,0 +1,75 @@
+{ buildPythonPackage
+, fetchPypi
+, lib
+, capstone
+, crytic-compile
+, ply
+, prettytable
+, psutil
+, pyelftools
+, pyevmasm
+, pysha3
+, pyyaml
+, rlp
+, setuptools
+, unicorn
+, wasm
+, wrapt
+, pkgs
+, yices
+, pytest
+, coverage
+}:
+
+let z3 = pkgs.z3;
+
+in buildPythonPackage rec {
+  pname = "manticore";
+  version = "0.3.4";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "trailofbits";
+    repo = "manticore";
+    rev = "4712491d379f1fa6c8faa2d38809877e31259f78";
+    sha256 = "0i9653013wvnj6y7f7i7vyj97250wd1qspr9ylpb1pnhi3c23gi0";
+  };
+
+  propagatedBuildInputs = [
+    pyyaml
+    wrapt
+    pysha3
+    prettytable
+    pyevmasm
+    rlp
+    ply
+    crytic-compile
+    wasm
+    psutil
+    setuptools
+    capstone
+    pyelftools
+    unicorn
+  ];
+
+  postPatch = ''
+    # Python API is not used in the code, only z3 from PATH
+    sed -ie s/z3-solver// setup.py
+    '';
+
+  checkInputs = [ pytest coverage ];
+  checkPhase = ''
+    export PATH=${yices}/bin:${z3}/bin:$PATH
+    # pytest tests/ethereum # TODO: enable when solc works
+    pytest tests/native -k 'not test_chmod'
+    pytest tests/other
+    pytest tests/wasm
+  '';
```

```suggestion
  checkInputs = [ pytestCheckHook ];
  preCheck = "export PATH=${yices}/bin:${z3}/bin:$PATH";
  pytestFlagsArray = [
    "--ignore=tests/ethereum"
  ];
  disabledTests = [ "test_chmod" ];
```
We usually don't bother running coverage in nixpkgs.